here-naksha-lib-handlers/src/test/java/com/here/naksha/lib/handlers/SourceIdHandlerUnitTest.java

```diff
@@ -0,0 +1,76 @@
+package com.here.naksha.lib.handlers;
+
+import com.here.naksha.lib.core.models.storage.NonIndexedPRef;
+import com.here.naksha.lib.core.models.storage.POp;
+import com.here.naksha.lib.core.models.storage.POpType;
+import org.junit.jupiter.api.Test;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class SourceIdHandlerUnitTest {
+
+    private static final String NS_COM_HERE_MOM_META = "@ns:com:here:mom:meta";
+
+    @Test
+    void tc2002_testMapEqToContainsTag() {
+        //given
+        NonIndexedPRef pRef = new NonIndexedPRef("properties", NS_COM_HERE_MOM_META, "sourceId");
+        POp given = POp.eq(pRef, "task_1");
+        //when
+
+        Optional<POp> result = SourceIdHandler.transformPopWithSourceId(given);
+        //then
+
+        assertTrue(result.isPresent());
+        assertEquals(result.get().getPropertyRef().getTagName(), "xyz_source_id_task_1");
+        assertEquals(result.get().op(), POpType.EXISTS);
+    }
+
+    @Test
+    void tc2003_testMapNotEqToNotContainsTag() {
+        //given
+        NonIndexedPRef pRef = new NonIndexedPRef("properties", NS_COM_HERE_MOM_META, "sourceId");
+        POp given = POp.not(POp.eq(pRef, "task_1"));
+        //when
+
+        Optional<POp> result = SourceIdHandler.transformPopWithSourceId(given);
+        //then
+
+        assertTrue(result.isPresent());
+        assertFalse(result.get().children().isEmpty());
+        POp nestedPop = result.get().children().get(0);
+
+        assertEquals(nestedPop.getPropertyRef().getTagName(), "xyz_source_id_task_1");
+        assertEquals(nestedPop.op(), POpType.EXISTS);
```

Comment:
nth / note: you can take a look at `POpAssertion` class from `service` module - it provides fluid assertion API for `POp`. Maybe it can get reused (it would have to moved to some shared dependency though). I'm leaving this to your judgement though, it's not that important

Reply:
I tried to introduce it but it appeared that it will require more dependencies work then skip this.

Reply:
sure